src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValueTaskExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
+{
+    internal static class ValueTaskExtensions
+    {
+        /// <summary>
+        /// Asserts the <see cref="ValueTask"/> passed has already been completed.
+        /// </summary>
+        /// <remarks>
+        /// This is useful for a specific case: sometimes you might be calling an API that is "sometimes" async, and you're
+        /// calling it from a synchronous method where you know it should have completed synchronously. This is an easy
+        /// way to assert that while silencing any compiler complaints.
+        /// </remarks>
+        public static void AssertCompleted(this ValueTask valueTask)
+        {
+            Contract.ThrowIfFalse(valueTask.IsCompleted);
+
+            if (valueTask.IsFaulted)
+            {
+                throw valueTask.AsTask().Exception;
+            }
```

Comment:
```suggestion
            // Observe the result, and throw the exception which occurred (if any)
            _ = valueTask.Result;
```

Reply:
This is ValueTask, not `ValueTask<T>`; it doesn't have a .Result?

Reply:
`GetAwaiter().GetResult()` would still work for this case. It's unnecessary for `ValueTask<T>` but required for `ValueTask`.